algebird-generic/src/main/scala/com/twitter/algebird/generic/JoinedAggregators.scala

```diff
@@ -0,0 +1,174 @@
+package com.twitter.algebird.generic
+
+import shapeless._
+import com.twitter.algebird._
+
+import Shapeless._
+
+/**
+ * This method allows to combine disparate aggregators that map to
+ * a unified presentation type.
+ *
+ * This diagram illustrates the "shape" of this combinator:
+ *
+ *   a1 -> b1 \
+ *   a2 -> b2 -> z
+ *   a3 -> b3 /
+ *
+ * Let's use the following example code to demonstrate the usage here:
+ *
+ *   val a: MonoidAggregator[Dog, DogStats, Result] = ...
+ *   val b: MonoidAggregator[Cat, CatStats, Result] = ...
+ *   val c: MonoidAggregator[Bird, BirdStats, Result] = ...
+ *   implicit val t: Semigroup[Result] = ...
+ *
+ * For products (things like tuples or heterogeneous lists) we can use
+ * .allOf to create a new aggregator:
+ *
+ *   val m1: Aggregator[Dog :: Cat :: HNil, DogStats :: CatStats :: HNil, Result] =
+ *     JoinedAggregators.allOf(a :: b :: HNil)
+ *
+ * For coproducts (things like Either or sealed traits) we can do the following:
+ *
+ *   val m2: MonoidAggregator[Dog :+: Cat :+: CNil, DogStats :: CatStats :: HNil, Result] =
+ *     JoinedAggregators.oneOf(a :: b :: HNil)
+ *
+ * ...and get a single aggregator that works for dogs or cats (wrapped in a coproduct)!
+ *
+ * We can also say:
+ *
+ *   val m3: MonoidAggregator[Animal, DogStats :: CatStats :: HNil, Result] =
+ *     JoinedAggregators(a :: b :: HNil).composePrepare(Generic[Animal].to)
```

Comment:
Same comment as above, why cant be implement this as a combinator using ApplicativeAggregators?

Reply:
Can you show me how? Suggest a change?